lib/events.js

```diff
@@ -306,7 +306,8 @@ EventEmitter.prototype.removeListener =
         else {
           delete events[type];
           if (events.removeListener)
-            this.emit('removeListener', type, listener);
+            this.emit('removeListener', type,
```

Comment:
I have a suggestion. Let's do `let currentListener = list.listener ? list.listener : listener` and check against it everywhere in this function. What do you think?